### ISSUE TITLE
onIndexChanged handler gets incorrect state

### ISSUE THREAD

**Issue by deepakaggarwal7** [leecade/react-native-swiper#938]
MAC, running on iOS emulator
- react-native-swiper": "^1.5.14
-  react-native": "0.57.5

### Expected behaviour
onIndexChanged should return the correct index.

### Actual behaviour
If I call setState in the onIndexChanged handler, it starts passing incorrect index intermittently. Note: If I do not call setState in the handler, it always returns right index. But I wonder why my setState is creating issues

I've also tried other options like onPagingRender or onScrollMomentumEnd etc. but same issue happens in all the moment I use setState in callback handler.
```
import React, {Component} from 'react'
import { View, Text, StyleSheet, Button } from 'react-native';
import Swiper from 'react-native-swiper';

export default class SwiperDemo extends Component {

    constructor(props){
        super(props)
         this.state = {selectedIndex:0}
    }
    updateState(index){
      console.log('index updated to ' + index);
      this.setState({
          selectedIndex:index
      })};
   
 

    render(){
      let curr=0;
       return(
            <View style={{flex:1, marginTop:30}}>
            <View style={{height:200}}>
            <Swiper showsButtons={false}
              onIndexChanged={this.updateState.bind(this)}>
              <View key={1} style={{height:300,backgroundColor:'red'}}><Text>Index should be 0</Text></View>
              <View key={2} style={{height:300,backgroundColor:'yellow'}}><Text>Index should be 1</Text></View>
              <View key={3} style={{height:300,backgroundColor:'blue'}}><Text>Index should be 2</Text></View>
            </Swiper>
            </View>
            <Text>Current Index in State = {this.state.selectedIndex}</Text>
            </View>
        )
    }
}

```
![ezgif com-resize](https://user-images.githubusercontent.com/19525740/53249671-b17e6100-36de-11e9-96f1-c78e321b9347.gif)

**Comment by EvanMaFYH** [comment]
same here

**Comment by retyui** [comment]
https://github.com/leecade/react-native-swiper/issues/713

**Comment by ArrayZoneYour** [comment]
merged #939. Thanks a lot 😊
